components/button/button-add.js

```diff
@@ -0,0 +1,131 @@
+import '../colors/colors.js';
+import '../icons/icon.js';
+import '../tooltip/tooltip.js';
+import { css, html, LitElement } from 'lit';
+import { FocusMixin } from '../../mixins/focus/focus-mixin.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { labelStyles } from '../typography/styles.js';
+import { LocalizeCoreElement } from '../../helpers/localize-core-element.js';
+
+/**
+ * A component for quickly adding items to a specific locaiton.
+ */
+class ButtonAdd extends FocusMixin(LocalizeCoreElement(LitElement)) {
+	static get properties() {
+		return {
+			/**
+			 * The text to be shown in the tooltip as a hint when visible-text is false
+			 * @type {string}
+			 */
+			label: { type: String },
+			/**
+			 * The text to show in the button when visible-text is true
+			 * @type {string}
+			 */
+			text: { type: String },
+			/**
+			 * When true, show the button with icon and visible text. When false, only show icon.
+			 * @type {boolean}
+			 */
+			visibleText: { type: Boolean, attribute: 'visible-text' }
```

Comment:
Most basic component states (demo page is also probably more useful for this):

1. Default -> `visible-text` is false, component shows the divider line, an add icon, and a tooltip which contains whatever is in `label` (design refers to this as "hint") OR "Add Item" as a fallback
2. Visible Text -> when `visible-text` is true, the component shows the divider, and in the centre is the plus icon and the text specified in `text` OR "Add Item" as a fallback.

`text` and `label` will not be used at the same time (the tooltip will not be shown if the text is visible and vice versa).

An aria description is potentially going to be used as well, I'll be discussing that more with design.